src/document/DocumentAnnotator.ts

```diff
@@ -106,10 +114,15 @@ export default class DocumentAnnotator extends BaseAnnotator {
             return;
         }
 
-        if (mode === Mode.HIGHLIGHT) {
-            this.annotatedEl.classList.add('ba-is-highlighting');
-        } else {
-            this.annotatedEl.classList.remove('ba-is-highlighting');
+        Object.values(ANNOTATION_CLASSES).forEach(className => {
+            if (this.annotatedEl && className) {
+                this.annotatedEl.classList.remove(className);
+            }
+        });
+
+        const className = ANNOTATION_CLASSES[mode];
+        if (className) {
```

Comment:
Added this because it's possible for className to be empty string in which case `.add` would not work.